app/frontend/wallet/shelley/transaction/computeTxPlan.ts

```diff
@@ -0,0 +1,268 @@
+import {InternalError, InternalErrorReason, UnexpectedErrorReason} from '../../../errors'
+import {CertificateType, Lovelace} from '../../../types'
+import {MAX_TX_OUTPUT_SIZE} from '../../constants'
```

Comment:
does this limit make sense anymore if we chose to put a fixed threshold on the amount of tokens in a single output?

Reply:
In a way yes, we still validate every output if its not too big (even tho they should not be) and we split outputs only if they surpass this value. Initially, I thought we might wanna split outputs regardless of the max size, but then I changed it so we split them only in that case. What do you think?

Reply:
I'd prefer to keep it simple - if the concern are fees, the user may save some by producing the bigger utxo now, but that would "kick them back" anyway if they choose to send some of the tokens out as the respective change output would be big.

So I don't see that as some improvement worth the additional complexity as it may not even be the most profitable strategy in the general case.

Also bigger outputs mean higher chances of issues with Trezor which still doesn't have unlimited throughput and/or interoperability with wallets which still struggle with big outputs.

Or did you see some other benefit that makes it worth to not break outputs down until they hit the global limit?

Reply:
My initial thought was that it "is ok to set a custom policy regarding change output splitting, but there is no need to set also custom `maxOutputSize` which we would kind of do if we split every output).  So basically my idea was to create a solution for an edge case, and don't influence also other cases if not necessary. Other than that I don't see any benefits to "not splitting all outputs based on token quatity".

Regarding the codebase, this is one condition we can easily remove so if you see any benefits to splitting every output (like the trezor support), lets do it.

Reply:
then I'd do the splitting by the fixed amount of max tokens per output all times - the logic is already quite hard to follow with all the edge-cases so I'd optimize for simplicity